web/ui/react-app/src/pages/rules/RulePanel.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { FC, CSSProperties, Fragment } from 'react';
+import { Rule } from '../../types/types';
+import { Link } from '@reach/router';
+import { createExpressionLink } from '../../utils';
+
+export const RulePanel: FC<{ rule: Rule; styles?: CSSProperties; tag?: keyof JSX.IntrinsicElements }> = ({
+  rule,
+  styles = {},
+  tag,
+}) => {
+  const Tag = tag || Fragment;
+  const { name, query, labels, annotations, alerts } = rule;
+  let customTagProps = {};
+  const mergedStyles = { background: '#f5f5f5', ...styles };
+  if (tag) {
+    customTagProps = { style: mergedStyles };
+  }
+  return (
+    <Tag {...customTagProps}>
+      <pre className="m-0" style={tag ? undefined : mergedStyles}>
+        <code>
+          <div>
+            {rule.alerts ? 'name: ' : 'record: '}
```

Comment:
- IMO better to check for `rule.type === 'alert'` here, as its more explicit.

- `name:` should be `alert:`